src/components/DayPickerRangeController.jsx

```diff
@@ -163,7 +163,7 @@ export default class DayPickerRangeController extends React.Component {
     if (focusedInput !== END_DATE) return false;
 
     if (startDate) {
-      const dayDiff = day.diff(startDate, 'days');
+      const dayDiff = day.day() - startDate.day();
```

Comment:
Nice. Yeah, let's skip doing it for day, but because of https://github.com/airbnb/react-dates/blob/master/src/utils/getCalendarMonthWeeks.js#L5, let's actually do 
```
day.diff(startDate.clone().startOf('day').hour(12), 'days');
```
just to cover all possible edge cases